libsmtutil/genz3wrapper.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+import sys
+import re
+
+# Patterns to match Z3 API entry point definitions.
+def_pat = re.compile(" *def_API(.*)")
+extradef_pat = re.compile(" *extra_API(.*)")
+# Pattern to extract name and arguments from the above.
+def_args_pat = re.compile("\('(.*)'[^\(\)]*\((.*)\)\s*\)")
+# Pattern to extract a list of arguments from the above.
+arg_list_pat = re.compile("[^\(]*\([^\)]*\)[, ]*")
+
+def generateEntryPoint(line, args):
+    m = def_args_pat.match(args)
+    if m:
```

Comment:
Switching to this would save you some indenting in the rest of the function:
```suggestion
    if !m:
        raise Exception(f'Could not parse entry point definition: {line}')
```